DOC: improve standard_t example in numpy.random.

lelayf opened the pull request:
This rewrite better reflects the cited reference R example
by offering a 2-tailed test and more accurate language.
Closes #18370.

<!--         ----------------------------------------------------------------
                MAKE SURE YOUR PR GETS THE ATTENTION IT DESERVES!
                ----------------------------------------------------------------

*  FORMAT IT RIGHT:
      http://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message

*  IF IT'S A NEW FEATURE OR API CHANGE, TEST THE WATERS:
      http://www.numpy.org/devdocs/dev/development_workflow.html#get-the-mailing-list-s-opinion

*  HIT ALL THE GUIDELINES:
      https://numpy.org/devdocs/dev/index.html#guidelines

*  WHAT TO DO IF WE HAVEN'T GOTTEN BACK TO YOU:
      http://www.numpy.org/devdocs/dev/development_workflow.html#getting-your-pr-reviewed
-->

Commit:
DOC: improve standard_t example in numpy.random.

This rewrite better reflects the cited reference R example
by offering a 2-tailed test and more accurate language.
Closes #18370.

rkern commented:
The docstring of [`Generator.standard_t`](https://github.com/numpy/numpy/blob/master/numpy/random/_generator.pyx#L1676) requires the same update.

lelayf commented:
I see, thank you for catching this. I will duplicate and update the PR.

bashtage commented on numpy/random/mtrand.pyx:
```diff
@@ -2137,6 +2137,9 @@ cdef class RandomState:
                Springer, 2002.
         .. [2] Wikipedia, "Student's t-distribution"
                https://en.wikipedia.org/wiki/Student's_t-distribution
+        .. [3] Walker, Helen M., "Degrees of freedom" in Journal of 
```
No change to RandomState should be made.  It is in maintenance-only mode.

lelayf replied:
Thank you for highlighting this, I was not aware. I will revert this file to its original state.

rkern replied:
Ehh, docstrings are a gray area that I didn't really consider in the NEP. We've allowed more significant code changes to `RandomState` since NEP 19. The core of the constraint in NEP 19 is that we don't change behaviors in `RandomState`, and this one doesn't.

bashtage commented on numpy/random/_generator.pyx:
```diff
@@ -1732,33 +1735,46 @@ cdef class Generator:
+        Does our t statistic land in one of the two critical regions found at 
+        both tails of the distribution?
 
         >>> np.sum(s<t) / float(len(s))
```
If a 2-sided test, it is better to use np.sum(np.abs(s) < np.abs(t))

lelayf replied:
Indeed! I Will update accordingly.

lelayf replied:
actually, if using abs, those values should be swapped, np.sum(np.abs(t) < np.abs(s))

bashtage commented on numpy/random/_generator.pyx:
```diff
@@ -1722,6 +1722,9 @@ cdef class Generator:
                Springer, 2002.
         .. [2] Wikipedia, "Student's t-distribution"
                https://en.wikipedia.org/wiki/Student's_t-distribution
+        .. [3] Walker, Helen M., "Degrees of freedom" in Journal of 
```
Should probably not be added.  This is not a well-cited article (109 only).

```
Degrees of freedom.
HM Walker - Journal of Educational Psychology, 1940 - psycnet.apa.org
The concept of degrees of freedom, considered of central importance to interpretation of
psychological and educational data, is presented and the computational procedure outlined.
A section is devoted to the importance of the concept and illustrations are given to show how …
  Cited by 109 Related articles All 13 versions 
```

bashtage reviewed (changes requested):
The big change is that RandomState should not be updated per the NEP.

Commit:
Remove poorly cited reference
Clarify point about std computation
Use absolute value for p-value computation of 2-tailed test

lelayf commented:
Latest commit keeps that docstring in sync for both Generator and RandomState. I addressed all comments above and carried a clarification regarding the use of `ddof=1` in `std`.

lelayf committed:
Merge branch 'master' into random-standard_t-doc

bashtage commented:
Looks ready to merge @mattip

bashtage reviewed (approved):
Looks ready.

charris commented:
Thanks @lelayf .

lelayf commented:
Great, thank you. First `numpy` PR for me, woohoo!
